public/app/features/alerting/unified/components/MenuItemPauseRule.tsx

```diff
@@ -39,48 +39,23 @@ const MenuItemPauseRule = ({ rule, onPauseChange }: Props) => {
     if (!isGrafanaRulerRule(rule.rulerRule)) {
       return;
     }
-    const ruleUid = rule.rulerRule.grafana_alert.uid;
-    const targetGroup = await getGroup({
-      rulerConfig: grafanaRulerConfig,
-      namespace: rule.namespace.uid || rule.rulerRule.grafana_alert.namespace_uid,
-      group: rule.group.name,
-    }).unwrap();
 
-    if (!targetGroup) {
-      notifyApp.error(
-        `Failed to ${newIsPaused ? 'pause' : 'resume'} the rule. Could not get the target group to update the rule.`
-      );
+    try {
+      const ruleGroupId = getRuleGroupLocationFromCombinedRule(rule);
+      await pauseRule(ruleGroupId, rule.rulerRule, newIsPaused);
+    } catch (error) {
+      notifyApp.error(`Failed to ${newIsPaused ? 'pause' : 'resume'} the rule: ${stringifyErrorLike(error)}`);
       return;
     }
 
-    // Parse the rules into correct format for API
-    const modifiedRules = targetGroup.rules.map((groupRule) => {
-      if (!(isGrafanaRulerRule(groupRule) && groupRule.grafana_alert.uid === ruleUid)) {
-        return groupRule;
-      }
-      return produce(groupRule, (updatedGroupRule) => {
-        updatedGroupRule.grafana_alert.is_paused = newIsPaused;
-      });
-    });
-
-    const payload = {
-      interval: targetGroup.interval!,
-      name: targetGroup.name,
-      rules: modifiedRules,
-    };
-
-    await updateRule({
-      nameSpaceUID: rule.namespace.uid || rule.rulerRule.grafana_alert.namespace_uid,
-      payload,
-    }).unwrap();
-
     onPauseChange?.();
   };
 
   return (
     <Menu.Item
       label={title}
       icon={icon}
+      disabled={updateState.isLoading}
```

Comment:
(Just pointing out that this doesn't do anything in practice as you can't see the menu once you click on it)

Reply:
we should disable the button, as while the rule is being paused, the UI let's user to click pause again